### PR TITLE
fix: add hint for missing `document` global in terminal error

### PR DESCRIPTION
**Pull request by bartlomieju**
This came up on Discord as a question so I thought it's worth adding a hint for this
as it might be a common pitfall.

### CODE REVIEW

**Comment by bartlomieju** [comment]
DO NOT LAND.

It's not great if `document` is used in testing code:
```
Deno.test("document query selector", () => {
  document.querySelector("div");
});
```

```
running 1 test from ./tests/specs/run/document/document_test.js
document query selector ... FAILED (1ms)

 ERRORS

document query selector => ./tests/specs/run/document/document_test.js:1:6
error: ReferenceError: document is not defined
  document.querySelector("div");
  ^
    at file:///Users/ib/dev/deno/tests/specs/run/document/document_test.js:2:3

 FAILURES

document query selector => ./tests/specs/run/document/document_test.js:1:6

FAILED | 0 passed | 1 failed (6ms)

error: Test failed
```

It should be enhancing terminal errors in `deno test` too.

**Comment by EGAMAGZ** [comment]
It will also grate to add this in the documentation related to testing, the need to add a library like `happy-dom`, `deno_dom`, `linkedom` or `JSDom` to polyfill the absence of dom to this type of test done with libraries like `@testing-library/preact` and frontend like `preact` (even include an example if it is possible)

File: runtime/fmt_errors.rs [hunk 1]
```diff
@@ -367,6 +367,14 @@ fn get_suggestions_for_terminal_errors(e: &JsError) -> Vec<FixSuggestion> {
           ]
         )
       ];
+    } else if msg.contains("document is not defined") {
+      return vec![
+        FixSuggestion::info("document global is not available in Deno."),
+        FixSuggestion::hint_multiline(&[
+          "Use a library like `happy-dom`, `deno_dom`, `linkedom` or `JSDom`",
+          "and setup `document` global according to the library documentation.",
```
Comment (dsherret) [inline runtime/fmt_errors.rs:375]:
```suggestion
          "Use a library like `happy-dom`, `deno_dom`, `linkedom` or `JSDom` and",
          "setup the `document` global according to the library documentation.",
```

Maybe

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**